app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,61 @@
+class CommentsController < ApplicationController
+    def index
```

Comment:
```suggestion
    before_action :set_comment, only: %i[show edit update destroy]

    def index
```
you are not using set_comment method you created.